kube-runtime/src/lib.rs

```diff
@@ -38,3 +38,34 @@ pub use watcher::{metadata_watcher, watcher};
 #[cfg(feature = "unstable-runtime-predicates")]
 pub use utils::{predicates, Predicate};
 pub use wait::conditions;
+
+pub mod prelude {
+    //! A "prelude" for kube runtime crate. Reduces the number of duplicated imports.
+    //!
+    //! This prelude is similar to the standard library's prelude in that you'll
+    //! almost always want to import its entire contents, but unlike the
+    //! standard library's prelude you'll have to do so manually:
+    //!
+    //! ```
+    //! use kube_runtime::prelude::*;
+    //! ```
+    //!
+    //! The prelude may grow over time as additional items see ubiquitous use.
+
+    pub use crate::reflector::Lookup as _;
```

Comment:
Lookup is an optimization trait used by reflector stores and generally no one else. Exporting this would lead to clashing names (Lookup exports the same names as Resource).

Reply:
Also planning on lifting this trait to other crates, would prefer not to export it for now.

Reply:
BTW, unrelated thought - `Lookup` trait is interesting, as it seems it can be a blanket trait implementation for `ResourceExt`. I see it as a cheap way to get resource to `ObjectReference` (for reference fetching) without extra hops, and if it will be declared as this:

```rust
/// Minimal lookup behaviour needed by a [reflector store](super::Store).
///
/// This trait is blanket-implemented for all [`Resource`] objects.
pub trait Lookup: ResourceExt {
    /// Constructs an [`ObjectRef`] for this object.
    fn to_object_ref(&self, dyntype: Self::DynamicType) -> ObjectRef<Self> {
        ObjectRef {
            dyntype,
            name: self.meta().name.clone().expect(".metadata.name missing"),
            namespace: self.meta().namespace.clone(),
            extra: Extra {
                resource_version: self.resource_version(),
                uid: self.uid(),
            },
        }
    }
}

impl<K: ResourceExt> Lookup for K {}
```
A couple of clones and a breaking change, but it seemingly compiles 🤷🏻